### PR TITLE
GLTFLoader + DRACOLoader: Fix bugs in glTF+Draco decompression.

### PR DESCRIPTION
**Pull request by donmccurdy**
Fixes https://github.com/mrdoob/three.js/issues/15039.

Fixes https://github.com/mrdoob/three.js/issues/15091.

Fixes https://github.com/KhronosGroup/glTF/issues/1267#issuecomment-431082451.

### CODE REVIEW

**Comment by donmccurdy** [comment]
Still working on a couple issues, consider this a WIP... 

https://github.com/mrdoob/three.js/issues/15091#issuecomment-431440737

File: examples/js/loaders/GLTFLoader.js [hunk 1]
```diff
@@ -2444,13 +2465,7 @@ THREE.GLTFLoader = ( function () {
...
 
 					// Use DRACO geometry if available
-					geometryPromise = extensions[ EXTENSIONS.KHR_DRACO_MESH_COMPRESSION ]
-						.decodePrimitive( primitive, parser )
-						.then( function ( geometry ) {
-
-							return addPrimitiveAttributes( geometry, primitive, parser );
-
-						} );
+					geometryPromise = createDracoPrimitive( primitive );
```
Comment (donmccurdy) [inline examples/js/loaders/GLTFLoader.js]: This was the main issue. `var` is not block scoped, unlike `let`, so by the time the callback returns, the value of `primitive` will have changed. All primitives were loaded using the last primitive definition. 😓

https://codeburst.io/asynchronous-code-inside-an-array-loop-c5d704006c99

**Comment by donmccurdy** [comment]
This is ready for review. :)

File: examples/js/loaders/GLTFLoader.js (hunk 1 above)
Comment (mrdoob) [reply examples/js/loaders/GLTFLoader.js]: 😁

**Comment by mrdoob** [comment]
Merging for now.

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**